debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerStreamingChangeEventSource.java

```diff
@@ -0,0 +1,263 @@
+        this.transactionalBufferMetrics.register(LOGGER);
+
+        this.transactionalBuffer = new TransactionalBuffer(connectorConfig.getLogicalName(), errorHandler, transactionalBufferMetrics, connectorConfig.getMaxQueueSize());
+
+        this.logMinerMetrics = new LogMinerMetrics(taskContext);
+        this.logMinerMetrics.register(LOGGER);
+
+        // The top outer loop gives the resiliency on the network disconnections. This is critical for cloud deployment.
+        while (context.isRunning()) {
+            try (Connection connection = jdbcConnection.connection(false);
+                    PreparedStatement fetchFromMiningView = connection
+                            .prepareStatement(SqlUtils.queryLogMinerContents(connectorConfig.getSchemaName(), jdbcConnection.username(), schema))) {
+
+                startScn = offsetContext.getScn();
+                createAuditTable(connection);
+                LOGGER.trace("current millis {}, db time {}", System.currentTimeMillis(), getTimeDifference(connection));
+                transactionalBufferMetrics.setTimeDifference(new AtomicLong(getTimeDifference(connection)));
+
+                if (!isContinuousMining && startScn < getFirstOnlineLogScn(connection)) {
+                    throw new RuntimeException("Online REDO LOG files don't contain the offset SCN. Clean offset and start over");
+                }
+
+                // 1. Configure Log Miner to mine online redo logs
+                setNlsSessionParameters(jdbcConnection);
+                checkSupplementalLogging(jdbcConnection, connectorConfig.getPdbName());
+
+                if (strategy == OracleConnectorConfig.LogMiningStrategy.CATALOG_IN_REDO) {
+                    buildDataDictionary(connection);
+                }
+
+                if (!isContinuousMining) {
+                    setRedoLogFilesForMining(connection, startScn);
+                }
+
+                LogMinerQueryResultProcessor processor = new LogMinerQueryResultProcessor(context, logMinerMetrics, transactionalBuffer,
+                        dmlParser, offsetContext, schema, dispatcher, transactionalBufferMetrics, catalogName, clock);
+
+                // 2. Querying LogMiner view while running
+                Set<String> currentRedoLogFiles = getCurrentRedoLogFiles(connection, logMinerMetrics);
+                while (context.isRunning()) {
+
+                    endScn = getEndScn(connection, startScn, logMinerMetrics);
+                    // LOGGER.trace("startScn: {}, endScn: {}", startScn, endScn);
+
+                    metronome = Metronome.sleeper(Duration.ofMillis(logMinerMetrics.getMillisecondToSleepBetweenMiningQuery()), clock);
+                    metronome.pause();
+
+                    Set<String> possibleNewCurrentLogFile = getCurrentRedoLogFiles(connection, logMinerMetrics);
+                    if (!currentRedoLogFiles.equals(possibleNewCurrentLogFile)) {
```

Comment:
If I understand correctly the check is only done before doing the work. So if the amount of records to process is huge the switch can happen during the processing. I always check the switch every 10 seconds.

Reply:
Added as a follow-up under https://issues.redhat.com/browse/DBZ-2457.

Reply:
I think it is also good idea, but checking before mining works too

Reply:
Only if the mine does not take very long. If the connector has been down for some time the amount of files to mine if increasing and so is the time it takes to handle them all. Chances are that during the mining the redo log rotates. At least, that is what I see in my environment. So I check every 10 seconds.

Reply:
This is a valid concern, however after a switch, the log still has all the contents to fetch. So, rotation should not be a problem unless this is the last redo group to rotate and has to be cleaned and reused. We recommend 7 groups (0.5G each file), so assuming each group will last for 10 minutes, we a good for 1 hour. Anyway checking every 10 seconds should be more secure.

Reply:
Ok. That is different from the setup I'm working with. Here sometimes the logs rotate within 10 seconds and we only have three files. This is our TST environment. In PRD we have 5 groups of 1G each that last a lot longer each. Still I prefer to check every couple of seconds. Costs are very low and safety gain is high.